### PR TITLE
INC-1001 Update only the current field for current incentives rather than whole record.

### PR DESCRIPTION
**Pull request by Mjwillis**
Signed-off-by: mjwillis <michael.willis@digital.justice.gov.uk>

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/incentivesapi/service/PrisonerIepLevelReviewService.kt [hunk 1]
```diff
@@ -66,13 +66,11 @@ class PrisonerIepLevelReviewService(
     return buildIepSummary(reviews, prisonApiService.getIncentiveLevels())
   }
 
-  @Transactional
```
Comment (Mjwillis) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/incentivesapi/service/PrisonerIepLevelReviewService.kt:69]: Don't want the creation and IEP and the Event to be in same transaction

File: src/main/kotlin/uk/gov/justice/digital/hmpps/incentivesapi/service/PrisonerIepLevelReviewService.kt [hunk 2]
```diff
@@ -136,19 +134,16 @@ class PrisonerIepLevelReviewService(
       throw ValidationException("Please provide fields to update")
     }
 
-    var prisonerIepLevel: PrisonerIepLevel? = prisonerIepLevelRepository.findById(id)
-    if (prisonerIepLevel == null) {
-      log.debug("PrisonerIepLevel with ID $id not found")
-      throw NoDataFoundException(id)
-    }
+    var prisonerIepLevel = prisonerIepLevelRepository.findById(id) ?: throw NoDataFoundException(id)
```
Comment (Mjwillis) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/incentivesapi/service/PrisonerIepLevelReviewService.kt:137]: This code is in a lot of places but just changed this here for now.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/incentivesapi/service/PrisonerIepLevelReviewService.kt [hunk 3]
```diff
@@ -367,14 +362,15 @@ class PrisonerIepLevelReviewService(
     return newIepReview
   }
 
-  private suspend fun persistIepLevel(
+  @Transactional
```
Comment (Mjwillis) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/incentivesapi/service/PrisonerIepLevelReviewService.kt:365]: Want to commit this before raising the event

**Review by andymarke — APPROVED** [review]
lgtm

**Pull request merged**